File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for(int i=0;i<nums.size();i++){
            int start = i;
            int end = nums.size()-1;
            while(start < end){
                if(nums[start]+nums[end]==target){
                    return {start, end};
                }
                end--;
            }
        }
        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        

        helper = {}


        for i in range(len(nums)):

            needed = target - nums[i]

            if needed in helper:
                return [i, helper[needed]]
            
            helper[nums[i]] = i
        
